internal/integration/knot/network/subscriber.go

```diff
@@ -24,17 +24,34 @@ func NewMsgSubscriber(amqp *AMQP) Subscriber {
 
 func (ms *msgSubscriber) SubscribeToKNoTMessages(msgChan chan InMsg) error {
 	var err error
-	subscribe := func(msgChan chan InMsg, queue, exchange, kind, key string) {
+
+	subscribe := func(msgChan chan InMsg, queue, exchange, kind, key string) error {
+		err = ms.amqp.OnMessage(msgChan, queue, exchange, kind, key)
 		if err != nil {
-			return
+			return err
 		}
-		err = ms.amqp.OnMessage(msgChan, queue, exchange, kind, key)
+		return nil
+	}
+
+	err = subscribe(msgChan, queueName, exchangeDevice, exchangeTypeDirect, BindingKeyRegistered)
+	if err != nil {
```

Comment:
I need to return something in case there is an error, even using another function to analyze the error, I would need to check the return of this verification function to return the error or not, I couldn't think of another way to do this.

Reply:
OK.